packages/edit-post/src/components/header/style.scss

```diff
@@ -189,6 +189,10 @@
 	}
 }
 
+.edit-post-header-toolbar__left > * + * {
```

Comment:
This is a really obtuse selector. What exactly does it do, and is there a CSS class you can target instead?

Reply:
It is, but there's a bit of a tangled web going on in that code. I'll loop @tellthemachines in here as I do think there's room for iteration and maybe simplification here. Spoiler, there is always room for reducing! She was incredible and helped me get to that point of the tangled web.

Reply:
This is the (somewhat unfortunately named) [lobotomized owl selector](https://alistapart.com/article/axiomatic-css-and-lobotomized-owls/). It targets all sibling elements except the first ( `* + *` meaning literally "any element that comes after another element"). Useful here because by adding a left margin to all but the first element, we ensure there's no extra margin at either end of the toolbar. Though it's possible to do the same with classes (`.classname + .classname`, we have a few of those sprinkled around the codebase), in this case the children of `edit-post-header-toolbar__left` don't all have the same class, and are not even all the same element, so `*` is the cleanest way to target them all.

Perhaps a comment explaining the selector would be helpful here? Something like `Select all but the first direct children of .edit-post-header-toolbar__left`?

Reply:
Because this rule is specific to the toolbar, it should probably live inside `packages/edit-post/src/components/header/header-toolbar/style.scss` instead. Also, it needs to be scoped to the text-only option, because currently the margin is being added on the icon buttons too. So I reckon the best place to put it is inside the `.show-icon-labels` rule on line 143 of the header-toolbar file.